### PR TITLE
NAS-129198 / 24.04.2 / Share "enabled" checkbox on edit right panel not matched share page

### PR DESCRIPTION
**Pull request by bvasilenko**
Restored as a backport. 
Original PR: https://github.com/truenas/webui/pull/10218

**Testing**

**Steps to Reproduce**

        
1. Navigate to Shares page

        
2. toggle share 'enabled' slider to 'disabled'

        
3. At this share click 'pencil' icon

**Expected Result**

right panel displays and 'enabled' checkbox is disabled

### CODE REVIEW

**Comment by undsoft** [comment]
Let's retarget this to just `master` and not have it in dragonfish.

File: src/app/pages/sharing/components/shares-dashboard/smb-card/smb-card.component.ts [hunk 1]
```diff
@@ -82,6 +83,7 @@ export class SmbCardComponent implements OnInit {
           iconName: 'edit',
           tooltip: this.translate.instant('Edit'),
           onClick: (row) => this.openForm(row),
+          disabled: (row) => this.changeEnabledIds$.pipe(map((ids) => ids.has(row.id))),
```
Comment (denysbutenko) [inline src/app/pages/sharing/components/shares-dashboard/smb-card/smb-card.component.ts:86]: I think you misunderstood the issue. The author means `disabled=unchecked` and `enabled=checked`. The problem is that the state is `unsynced` for toggle&checkbox between card and form.

**Review by denysbutenko — CHANGES_REQUESTED** [review]

**Comment by bvasilenko** [comment]
replaced by https://github.com/truenas/webui/pull/10218

**Review by denysbutenko — APPROVED** [review]
backported

**Pull request merged**